### PR TITLE
Upload a picture to user profile in Target MVD

### PR DESCRIPTION
**Pull request by afmicc**
#### Trello board reference:

* [As a user I should be able to upload a picture to my profile](https://trello.com/c/J2r6MLdV)

---

#### Description:

* Add active storage to save user avatar
* Add resize image to user avatar

### CODE REVIEW

**Review by Alxzu — APPROVED** [review]
This is fine, I'm wondering if you should change development to amazon too in order to test the feature since local seems to far away from the production mode, wdyt?

File: app/controllers/api/v1/users_controller.rb [hunk 1]
```diff
@@ -15,7 +17,7 @@ def user
       end
 
       def update_params
-        params.require(:user).permit(:name, :gender)
+        params.require(:user).permit(:name, :gender, avatar: %i[data filename content_type])
```
Comment (gabrielbursztein2) [inline app/controllers/api/v1/users_controller.rb]: Are you expecting a base64 file? If this image is used for an Avatar, why do you need filename and content_type?

File: spec/request/api/v1/users/show_spec.rb [hunk 2]
```diff
@@ -0,0 +1,68 @@
+require 'rails_helper'
+
+describe 'GET api/v1/user/id', type: :request do
+  let!(:user) { create(:user, :confirmed) }
+
+  subject { get api_v1_user_path(user), headers: auth_header }
+
+  context 'when the request is succesful and user has avatar' do
+    let(:avatar) do
```
Comment (gabrielbursztein2) [inline spec/request/api/v1/users/show_spec.rb]: Are you using this?

**Comment by afmicc** [comment]
> This is fine, I'm wondering if you should change development to amazon too in order to test the feature since local seems to far away from the production mode, wdyt?

@Alxzu You are right but I don't have amazon credentials yet.

File: Gemfile [hunk 3]
```diff
@@ -3,7 +3,9 @@ git_source(:github) { |repo| "https://github.com/#{repo}.git" }
 
 ruby '2.6.3'
 
+gem 'active_storage_base64', '~> 0.1.4'
 gem 'activeadmin', '~> 2.2'
+gem 'aws-sdk-s3', require: false
```
Comment (JuanfraM) [inline Gemfile]: version is missing here

File: Gemfile [hunk 4]
```diff
@@ -25,17 +28,14 @@ gem 'therubyracer', '~> 0.12.3'
...
-
 # Use Capistrano for deployment
 # gem 'capistrano-rails', group: :development
 
 # Reduces boot times through caching; required in config/boot.rb
 gem 'bootsnap', '>= 1.1.0', require: false
 
 # Use Rack CORS for handling Cross-Origin Resource Sharing (CORS), making cross-origin AJAX possible
-# gem 'rack-cors'
+gem 'rack-cors'
```
Comment (JuanfraM) [inline Gemfile]: version

File: spec/request/api/v1/users/show_spec.rb [hunk 5]
```diff
@@ -0,0 +1,60 @@
+require 'rails_helper'
+
+describe 'GET api/v1/user/id', type: :request do
+  let!(:user) { create(:user, :confirmed) }
+
+  subject { get api_v1_user_path(user), headers: auth_header }
+
+  context 'when the request is succesful and user has avatar' do
+    before do
```
Comment (JuanfraM) [inline spec/request/api/v1/users/show_spec.rb:9]: you can use the `trait` created in the user factory instead of attaching the avatar before each test

**Review by JuanfraM — APPROVED** [review]
Some minor comments.
Good job 💪

**Pull request merged**